gstd/src/lock/mutex.rs

```diff
@@ -193,18 +213,55 @@ impl<'a, T> Future for MutexLockFuture<'a, T> {
     // In case of locked mutex and an `.await`, function `poll` checks if the
     // mutex can be taken, else it waits (goes into *waiting queue*).
     fn poll(self: Pin<&mut Self>, _cx: &mut Context<'_>) -> Poll<Self::Output> {
-        let current_msg_id = crate::msg::id();
-        let lock = unsafe { &mut *self.mutex.locked.get() };
-        if lock.is_none() {
-            *lock = Some(current_msg_id);
+        let current_msg_id = msg::id();
+        let current_block_number = exec::block_height();
+        let locked_by = unsafe { &mut *self.mutex.locked.get() };
+        if let Some((lock_owner_msg_id, deadline_block_number)) = *locked_by {
+            if current_block_number > deadline_block_number {
+                if let Some(msg_future_task) = async_runtime::futures().get_mut(&lock_owner_msg_id)
```

Comment:
I do not really like this dependency on the `async_runtime` here. Probably the better option would be having a separate list of ousted messages inside the `Mutex` struct. Every ousted message gets moved into there and awoken. During the `poll` the future can check its message in that list and panic here instead of the main message loop. Any thoughts?